packages/@glimmer/manager/lib/internal/index.ts

```diff
@@ -138,7 +139,18 @@ export function getInternalHelperManager(
     );
   }
 
-  const manager = getManager(HELPER_MANAGERS, definition)!;
+  let manager = getManager(HELPER_MANAGERS, definition)!;
+
+  // Functions are special-cased because functions are defined
+  // as the "default" helper, per: https://github.com/emberjs/rfcs/pull/756
+  if (manager === undefined && typeof definition === 'function') {
+    setInternalHelperManager(
+      new CustomHelperManager(() => new FunctionHelperManager()),
+      definition
+    );
+
+    manager = getManager(HELPER_MANAGERS, definition)!;
```

Comment:
Do we really need to set the manager, as opposed to just returning it (well, assigning it to the variable here and 167 will do the returning)? Is this effectively just caching the lookup?

Also, is there a way to avoid making a new instance each time, since the manager seems entirely stateless?

Reply:
oh nice! I hadn't thought of that. updated